cprover_bindings/src/goto_program/typ.rs

```diff
@@ -166,6 +170,8 @@ impl DatatypeComponent {
             | Double
             | FlexibleArray { .. }
             | Float
+            | Float16
+            | Float128
```

Comment:
We have methods `is_float` and `is_double` - please create equivalents of those and inspect all use cases of `is_float` and `is_double` to see whether you have to add calls to the newly added methods.

Reply:
Added new methods for `f16` and `f128`